homeassistant/components/goalzero/const.py

```diff
@@ -25,6 +55,86 @@
     "inputDetected": ["Input Detected", DEVICE_CLASS_POWER, None],
 }
 
+SENSOR_DICT = {
+    "wattsIn": {
+        ATTR_NAME: "Watts In",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "ampsIn": {
+        ATTR_NAME: "Amps In",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_CURRENT,
+        ATTR_UNIT_OF_MEASUREMENT: ELECTRICAL_CURRENT_AMPERE,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: False,
+    },
+    "wattsOut": {
+        ATTR_NAME: "Watts Out",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "ampsOut": {
+        ATTR_NAME: "Amps Out",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_CURRENT,
+        ATTR_UNIT_OF_MEASUREMENT: ELECTRICAL_CURRENT_AMPERE,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: False,
+    },
+    "whOut": {
+        ATTR_NAME: "WH Out",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        ATTR_UNIT_OF_MEASUREMENT: ENERGY_WATT_HOUR,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: False,
+    },
+    "whStored": {
+        ATTR_NAME: "WH Stored",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        ATTR_UNIT_OF_MEASUREMENT: ENERGY_WATT_HOUR,
+        ATTR_STATE_CLASS: STATE_CLASS_MEASUREMENT,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "volts": {
+        ATTR_NAME: "Volts",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_VOLTAGE,
+        ATTR_UNIT_OF_MEASUREMENT: VOLT,
+        ATTR_DEFAULT_ENABLED: False,
+    },
+    "socPercent": {
+        ATTR_NAME: "State of Charge Percent",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_BATTERY,
+        ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "timeToEmptyFull": {
+        ATTR_NAME: "Time to Empty/Full",
+        ATTR_DEVICE_CLASS: TIME_MINUTES,
+        ATTR_UNIT_OF_MEASUREMENT: TIME_MINUTES,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "temperature": {
+        ATTR_NAME: "Temperature",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+        ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "wifiStrength": {
+        ATTR_NAME: "Wifi Strength",
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_SIGNAL_STRENGTH,
+        ATTR_UNIT_OF_MEASUREMENT: SIGNAL_STRENGTH_DECIBELS,
+        ATTR_DEFAULT_ENABLED: True,
+    },
+    "timestamp": {
+        ATTR_NAME: "Up Time",
+        ATTR_UNIT_OF_MEASUREMENT: TIME_SECONDS,
+        ATTR_DEFAULT_ENABLED: False,
+    },
```

Comment:
Meh, I don't like myself now. Just after merging this PR, I noticed this one.
It is not allowed for a sensor to have an uptime in seconds, instead, it should be a timestamp of when it came online.

Reply:
This is the total run time o the device which may or may not reset during a factory reset. Should it just be renamed?

Reply:
Hmm... ok well the key is called `timestamp`, while the name is "Up Time", however it is "Total ru time"? 😉 

Hmz yeah, if it is run time, it maybe should be called that.

Reply:
If possible it would be better to calculate the timestamp of the start by deducting the run time from the current time, possibly rounding off to minutes. Then use device class timestamp.